### PR TITLE
Chore: more type fixes

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- Fixes some more `any`/type assertions

**Why do we need this feature?**

- better types across grafana

**Who is this feature for?**

- grafana devs!

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/plugins/datasource/testdata/components/SimulationQueryEditor.tsx [hunk 1]
```diff
@@ -31,7 +31,7 @@ export const SimulationQueryEditor = ({ onChange, query, ds }: EditorProps) => {
 
   // This only changes once
   const info = useAsync(async () => {
-    const v = (await ds.getResource('sims')) as SimInfo[];
+    const v: SimInfo[] = await ds.getResource('sims');
```
Comment (joshhunt) [inline public/app/plugins/datasource/testdata/components/SimulationQueryEditor.tsx:34]: use `getResource` generic instead `const v = await ds.getResource<SimInfo[]>('sims');`

**Review by joshhunt — COMMENTED** [review]
just a few minor nits :)

**Review by itsmylife — APPROVED** [review]
LGTM 🚀

**Review by mckn — APPROVED** [review]
LGTM

**Pull request merged**